### DISCUSSION TITLE
Z2M "Dashboard" tab not showing

### DISCUSSION THREAD

**Discussion by ovizii** [Koenkk/zigbee2mqtt#12814] [General]
I'm busy setting up HA on a Raspberry Pi for a holiday home so having it up and running, I added 1 device to Z2M and I can see it, albeit with a few problems. It is this device: https://www.zigbee2mqtt.io/devices/HS2WD-E.html#heiman-hs2wd-e  

Any advice?

![image](https://user-images.githubusercontent.com/837902/173413037-cc011ed7-ba84-4c3a-a342-fc66bd04d091.png)

It seems to have a problem or two:

![image](https://user-images.githubusercontent.com/837902/173413146-a991254d-f5e7-4304-bef8-e8af807a7ea1.png)

![image](https://user-images.githubusercontent.com/837902/173413580-f3290980-213e-488d-914f-91a4f785fad7.png)

![image](https://user-images.githubusercontent.com/837902/173413243-6ded8f48-550f-4157-8e62-762c9b857ae8.png)

Logs:
```
2022-06-13 19:14:44 WARNING (MainThread) [homeassistant.components.mqtt.mixins] Erroneous JSON: 
2022-06-13 19:14:44 ERROR (MainThread) [homeassistant.helpers.template] Template variable error: 'value_json' is undefined when rendering '{{ value_json.battery }}'
```

AND the "dashboard" is blank, Google Chrome isn't showing any errors so I dunno. 
![image](https://user-images.githubusercontent.com/837902/173413406-33da0a9e-cd72-4293-a964-03c351d37ab1.png)

Open for any suggestions to debug this mess :-/

**Comment by ovizii** [comment]
btw. I paired this [HS2WD-E](https://www.zigbee2mqtt.io/devices/HS2WD-E.html#heiman-hs2wd-e) device with another Z2M instance which has several paired devices and the dashboard works and shows all devices except this one, so it's not a general dashboard problem but a problem with HS2WD-E not showing up in the dashboard.

Any ideas?

**Comment by sjorge** [comment]
I think there are no exposes on this device that will trigger the creation of a dashboard card.

**Reply by ovizii** [reply]
Oh, I see, so that is what the “empty exposes definition” means. Thanks for sharing that with me.

**Comment by ovizii** [comment]
I found another device which is not showing in Z2M's dashboard, although it is listed as device and exposes stuff. Any ideas why and is there anything to do about this?

![image](https://user-images.githubusercontent.com/837902/190425672-3ead9688-7a25-4fe3-823d-43d0a5301c67.png)

![image](https://user-images.githubusercontent.com/837902/190425835-83d574b2-28e7-425e-b107-262b0dedb15d.png)

**Comment by danvy** [comment]
The Aqara wireless remote switch H1 [WXKG15LM] has the same problem. The device exposes several properties but doesn't show up in the dashboard tab. Other supported Aqara devices are there.

**Reply by edrzns** [reply]
Same with LYWSD03MMC temp/humidity sensors re-flashed with Zigbee firmware

**Comment by HAOS-DGRobert** [comment]
Hi,

I have the same problem. I recently bought several Heiman HS1SA smoke sensors.  I enrolled them in Z2M but they appear as HS3SA, a newer model I guess.  They all appear as devices but only 2 appear in the Z2M Dashboard. I noticed that they appeared after a restart at the Host.  Strangely, the other sensors completely disappeared from the Z2M and I had to re-enroll them but also still not showing in Z2M Dashboard. Any Ideas?

**Comment by jolexa** [comment]
I found this discussion via Google Search. I can't quite figure out what triggers dashboard card or not. For example, I have a few [contact sensors](https://www.zigbee2mqtt.io/devices/3RDS17BZ.html) that only expose contact status and I cannot set anything manually -- they are on the dashboard. However, [the Button](https://www.zigbee2mqtt.io/devices/3RSB22BZ.html), which also are "read only" -- do **not** show in the dashboard. It is inconsistent enough that I can't pinpoint what is the trigger. Seems like a bug??

**Comment by gkfirst8** [comment]
Looks like I'm in the same situation: my "Innr SP 240" does show up in the Dashboard but my "Robb smarrt" switch does not. I'd love to learn about this. Does it have to do with the order in which it was added? In my case the "Robb smarrt" switch was not supported yet and I actually made a PR to add it. (Which got accepted quickly.) Maybe I will re-pair the switch to see if it then gets added to the Dashboard.

**Comment by gkfirst8** [comment]
OK, even after re-pairing it does not show up.

Can we manually add stuff to the Dashboard? Maybe through some MQTT-message? Oh, wait, the UI/Dashboard is a different repo/project, isn't it?

**Comment by oywino** [comment]
Utterly sad that  nobody could answer this question......

**Comment by xbrunosousa** [comment]
Same here with IKEA E2001/E2002.
